src/lib/profiles/software-update/SoftwareUpdateProfile.h

```diff
@@ -59,357 +59,389 @@ namespace nl {
-   * and message type 1. the structure provided here is not really used
-   * except as a placeholder.
-   */
-  class ImageAnnounce { };
-  /*
-   * the image query frame has the following form.
-   *
-   * | <frame control> | <product specification> | <vendor specific data> (optional) |
-   * | <version specification> | <locale specification> (optional) |
-   * | <integrity type list> | <update scheme list > |
-   *
-   * where the frame control field has bit-fields as follows:
-   * bit 0 - 1==vendor-specifc data present, 0==not present
-   * bit 1 - 1==locale specification present, 0==not present
-   *
-   * the image query, as a structure reads slightly differently from the in-flight
-   * representation. in particular, the version and locale are null-terminated
-   * strings and both of the optional items are represented as nullable pointers
-   * so there isn't a separate boolean to check.
-   *
-   * here's an auxiliary class that holds a list of integrity types as part of
-   * the image query. these are pretty simple and are essentially wrappers
-   * around a 3-element array of values drawn from the above enumeration.
-   */
-  class NL_DLL_EXPORT IntegrityTypeList {
-  public:
+/**
+ * An auxiliary class to hold a list of integrity types as a part of the image query.
+ *
+ * A simple wrapper, sized to hold any subset of possible integrity types. In
+ * order to accomplish this task, its size is equal to the number of elements in
+ * the @ref IntegrityTypes.  It is used to generate the list of supported
+ * integrity types in the ImageQuery message.
+ */
+class NL_DLL_EXPORT IntegrityTypeList
+{
+public:
     // constructor
     IntegrityTypeList();
     // initializer
-    WEAVE_ERROR init(uint8_t, uint8_t*);
+    WEAVE_ERROR init(uint8_t, uint8_t *);
     // packing and parsing
-    WEAVE_ERROR pack(MessageIterator&);
-    uint16_t packedLength();
-    static WEAVE_ERROR parse(MessageIterator&, IntegrityTypeList&);
+    WEAVE_ERROR pack(MessageIterator &);
+    static WEAVE_ERROR parse(MessageIterator &, IntegrityTypeList &);
     // comparison
-    bool operator == (const IntegrityTypeList&) const;
-    // this is just a short list with specified length. if length is
-    // 0 this indicates that the list is undefined since, to be useful,
-    // one of these things has to caontain at least one element.
-    uint8_t theLength;
-    uint8_t theList[3];
-  };
-  /*
-   * an auxiliary class that holds a list of update schemes as part of
-   * the image query. these are pretty simple and are essenitlly wrappers
-   * around a 4-element array of values drawn from the above enumeration.
-   */
-  class NL_DLL_EXPORT UpdateSchemeList {
-  public:
+    bool operator ==(const IntegrityTypeList &) const;
+    uint8_t theLength;  /**< Length of the supported element list.  Length of 0 indicates an empty list */
+    uint8_t theList[3]; /**< Container holding supported integrity types.  It is sized equal to the number of elements in @ref
```

Comment:
3: magic number -> change to a mnemonic used here and in the source code?

Reply:
done